### PR TITLE
reg: Fix aarch64-darwin build

### PR DESCRIPTION
**Pull request by ereslibre**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/reg/default.nix [hunk 1]
```diff
@@ -14,6 +16,14 @@ buildGoModule rec {
     sha256 = "1jlza1czfssssi3y9zi6kr8k9msfa7vp215ibhwbz4h97av5xw5m";
   };
 
+  patches = lib.optionals (stdenv.hostPlatform.system == "aarch64-darwin") [
+    (fetchpatch {
+      name = "fix-aarch64-darwin.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/genuinetools/reg/pull/218.patch";
+      hash = "sha256-mSahDRcGYD1RWiK0FhmeXUMWm9SorFrTKPn4QxdQL0U=";
+    })
```
Comment (tjni) [inline pkgs/tools/virtualization/reg/default.nix:24]: Could we fetch the commits from your PR directly. This way, it doesn't reference the pull request itself, which can change over time:

```
  patches = [
    # https://github.com/genuinetools/reg/pull/218
    (fetchpatch {
      name = "update-x-sys-for-go-1.18-on-aarch64-darwin.patch";
      url = "https://github.com/genuinetools/reg/commit/f37b04ad8be47f1e68ef9b2c5906324d7096231e.patch";
      hash = "sha256-wmBjPdrpNpxx//I+d+k8DNR11TmyXxb84vXR/MrYHKA=";
    })
    (fetchpatch {
      name = "update-vendored-dependencies.patch";
      url = "https://github.com/genuinetools/reg/commit/8bb04bc3fd41c089716e65ee905c6feac8bda5a0.patch";
      hash = "sha256-5nZaayy0xu7sOJHVCp6AotINiF7vXk2Jic8SR8ZrH/g=";
    })
  ];

File: pkgs/tools/virtualization/reg/default.nix [hunk 2]
```diff
@@ -14,6 +16,14 @@ buildGoModule rec {
     sha256 = "1jlza1czfssssi3y9zi6kr8k9msfa7vp215ibhwbz4h97av5xw5m";
   };
 
+  patches = lib.optionals (stdenv.hostPlatform.system == "aarch64-darwin") [
```
Comment (tjni) [inline pkgs/tools/virtualization/reg/default.nix:19]: Let's apply the patch unconditionally, since that will be the case if the PR is merged.

**Review by tjni — COMMENTED** [review]
Thanks for fixing this.

**Review by tjni — APPROVED** [review]
LGTM

**Comment by ereslibre** [comment]
Thanks @tjni!

**Review by figsoda — APPROVED** [review]

**Pull request merged**